schematics/src/step/files/generate-an-app/app.component.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AppComponent } from './app.component';
+
+describe('AppComponent', () => {
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule
+      ],
+      declarations: [
+        AppComponent
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ],
+    }).compileComponents();
+  });
+
+  it('should create the app', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect(app).toBeTruthy();
+  });
+
+  it(`should have as title 'place-my-order'`, () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect(app.title).toEqual('place-my-order');
+  });
+
+  it('should render title', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    fixture.detectChanges();
+    const compiled = fixture.nativeElement as HTMLElement;
+    expect(compiled.querySelector('h1')?.textContent).toContain('Place My Order App: Coming Soon!');
+  });
+});
```

Comment:
I haven't considered referring to files outside the `./files` folder, because by default schematics add files under `./files` schematic folder to the build of the schematic, thus it's available for both a dev environment and a production environment (when user is using the schematic).
With the current logic, we are not able to get files other than `./files` folder because we depend on schematic logic of the way it looks at a `source` folder and iterates over the files in it (`url` method). It does not provide a method to look at only one file outside of the schematic folder. The method `url` looks at a virtual file tree instead of the file system.

![image](https://github.com/bitovi/academy/assets/26363255/e8f986ea-a2c2-45b2-a485-426f1ad384bc)

I think looking at files outside of the schematic folder would diverge from schematic methods but it's not impossible, it just would not be following the schematics best practices to handle the file tree.
I think we could start with this setup and then figure out if it can be improved later.
The main thing at this point is that people updating references to files in the academy would have to be aware of updating the schematic file as well.